### PR TITLE
remove C++ coc (use pycoc)

### PR DESCRIPTION
**Pull request by skiars**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,17 +9,13 @@ INCPATH   = src default
...
+COC       = python3 tools/coc/main.py
 CONST_TAB = $(GENERATE)/be_const_strtab.h
-# MAKE_COC  = $(MAKE) -C tools/coc
 
 ifeq ($(OS), Windows_NT) # Windows
     CFLAGS    += -Wno-format # for "%I64d" warning
     LFLAGS    += -Wl,--out-implib,berry.lib # export symbols lib for dll linked
     TARGET    := $(TARGET).exe
-    # COC       := $(COC).exe
     PY3       := $(PY3).exe
```
Comment (s-hadinger) [inline Makefile]: This line will not work anymore without PY3 being defined.

Does Windows requires `python3.exe`?

Comment (skiars) [reply Makefile]: Then I will try it tomorrow.

File: Makefile [hunk 2]
```diff
@@ -1,26 +1,22 @@
...
 LFLAGS    = 
 
 INCPATH   = src default
 SRCPATH   = src default
 GENERATE  = generate
 CONFIG    = default/berry_conf.h
-# COC		  = tools/coc/coc
-PY3       = python3
-PYCOC     = tools/pycoc/main.py
+COC       = tools/coc/coc
```
Comment (s-hadinger) [inline Makefile:12]: Should be `COC       = tools/pycoc/main.py`

File: Makefile [hunk 3]
```diff
@@ -1,26 +1,22 @@
...
 CONST_TAB = $(GENERATE)/be_const_strtab.h
-# MAKE_COC  = $(MAKE) -C tools/coc
 
 ifeq ($(OS), Windows_NT) # Windows
     CFLAGS    += -Wno-format # for "%I64d" warning
     LFLAGS    += -Wl,--out-implib,berry.lib # export symbols lib for dll linked
     TARGET    := $(TARGET).exe
-    # COC       := $(COC).exe
-    PY3       := $(PY3).exe
+    COC       := python $(COC) # need python3
```
Comment (s-hadinger) [inline Makefile]: I think we need to enforce `python3`. On Mac and Linux, `python` often still points to `python 2.7`

I suggest `    COC       := python3 $(COC) # need python3`

**Review by s-hadinger — COMMENTED** [review]
Minor changes

File: Makefile (hunk 3 above)
Comment (skiars) [reply Makefile]: There is no python3 command on windows, you must install the correct version :-(

Comment (s-hadinger) [reply Makefile]: Arf... Unfortunately I also need Python 2 for legacy reasons... Then can we split and do `python.exe` on windows, and `python3` for the rest of the world?

File: tools/coc/coc [hunk 4]
```diff
@@ -1,5 +1,4 @@
-# python3 main.py -o out in
-# coc 
+#! /usr/bin/env python3
```
Comment (skiars) [inline tools/coc/coc:1]: In unix-like, you can directly use coc as an executable file, and it uses python3.

File: Makefile (hunk 3 above)
Comment (skiars) [reply Makefile]: You can look at the Makefile, this line is only for windows.

File: tools/coc/coc (hunk 4 above)
Comment (s-hadinger) [reply tools/coc/coc:1]: Oh yes, I missed that. Sorry

File: Makefile (hunk 3 above)
Comment (s-hadinger) [reply Makefile]: Then it's good to me :)

**Review by s-hadinger — APPROVED** [review]

**Review by s-hadinger — APPROVED** [review]

**Pull request merged**